test/benchmarks.ts

```diff
@@ -0,0 +1,141 @@
+    targetGitRef: "v1.1.0",
+    filePatterns: ["."],
+    browserslist: ["electron 7.1.13"],
+    eslintOptions: {
+      cwd: `${projectRoot}/benchmarks-tmp/electron-react-boilerplate`,
+      extensions: [".js", ".jsx", ".ts", ".tsx"],
+      useEslintrc: false,
+      baseConfig: {
+        extends: ["plugin:compat/recommended"],
+      },
+    },
+  },
+];
+
+async function getRepo({ remoteLink, location }: RepoInfo) {
+  // Clone if necessary, else use existing repo
+  return fs
+    .access(`${location}/.git`, fsConstants.R_OK)
+    .then(() => {
+      console.log(`Using existing ${location}`);
+      return Repository.open(location);
+    })
+    .catch(() => {
+      console.log(`Cloning ${remoteLink}`);
+      return Clone.clone(remoteLink, location);
+    });
+}
+
+async function editBrowserslistrc(path: string, val: Array<string>) {
+  const file = await fs.readFile(path);
+  const json = JSON.parse(file.toString());
+  json.browserslist = val;
+  await fs.writeFile(path, JSON.stringify(json));
+}
+
+async function getBenchmark(repoInfo: RepoInfo) {
+  const { name, remoteLink, targetGitRef, eslintOptions, location } = repoInfo;
+  console.log(`Retrieving ${remoteLink}`);
+  const repo = await getRepo(repoInfo);
+  const ref = await repo.getReference(targetGitRef);
+  const eslint = new ESLint(eslintOptions);
+  console.log(`Checking out ${name} ${targetGitRef}`);
+  await repo.checkoutRef(ref);
+  if (repoInfo.browserslist) {
+    const packageJsonPath = `${location}/package.json`;
+    console.log(`Editing browserslistrc in ${packageJsonPath}`);
+    await editBrowserslistrc(packageJsonPath, repoInfo.browserslist);
+  }
+
+  const benchmark = new Benchmark(
+    name,
+    (deferred: { resolve: Function }) => {
+      eslint
+        .lintFiles(location)
+        .then(() => {
+          return deferred.resolve();
+        })
+        .catch((e) => console.error(e));
+    },
+    {
+      onStart: () => {
+        console.log(`Starting ${name} ${targetGitRef} ESLint benchmark`);
+      },
+      onComplete: () => {
+        console.log(`Completed benchmark ${name}`);
+      },
+      onError: () => {
+        console.error(benchmark.error);
+      },
+      async: true,
+      defer: true,
+      maxTime: 30,
+    }
+  );
+  return benchmark;
+}
+
+(async function main() {
+  const benchmarks = repos.map((repo) => {
+    return getBenchmark(repo);
+  });
+
+  const resolvedBenchmarks = await Promise.all(benchmarks);
+  Benchmark.invoke(resolvedBenchmarks, {
+    name: "run",
+    async: true,
+    onStart: () => console.log(`Starting benchmark suite`),
+    onComplete: (e) => {
+      console.log("Finished benchmark suite");
+      const reports = e.currentTarget.map((benchmark) => ({
+        name: benchmark.name,
+        stats: benchmark.stats,
+        sample: benchmark.stats.sample,
+        sampleCount: benchmark.stats.sample.length,
+      }));
+      console.log(reports);
+    },
+  });
+})().catch((e) => {
```

Comment:
This will not fail on errors because `console.error` does not throw. See https://github.com/amilajack/alfred/blob/ebd4eda886eeb609957b2bca86c6217cbb8ea9bb/packages/cli/tests/e2e/alfred-e2e-start.ts#L9-L11 as suggestion for implementation